driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -686,6 +690,12 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) {
             }
         }
 
+        @Override
+        public void channelIdle(ChannelHandlerContext ctx, IdleStateEvent e) throws Exception {
```

Comment:
heartbeatInterval should be higher than readTimeout, I'm going to document that in the API.

But as you said it wouldn't be a problem, there are already cases where we close a connection while requests are still in fight, so the driver knows how to handle that.